File: message-bus/src/main/java/alma/icd/adapt/messagebus/AbstractMessage.java
package alma.icd.adapt.messagebus;

import com.fasterxml.jackson.annotation.JsonIgnore;

import alma.icd.adapt.messagebus.Envelope;
import alma.icd.adapt.messagebus.Message;

public abstract class AbstractMessage implements Message {

	// Need to ignore envelope when (de)serializing -- Jackson cannot
	// cope with circular references like Envelope -> Message -> Envelope
	@JsonIgnore
	private Envelope envelope;

	@Override
	public Envelope getEnvelope() {
		return this.envelope;
	}

	@Override
	public void setEnvelope(Envelope envelope) {
		this.envelope = envelope;
	}
}

File: message-bus/src/main/python/adapt/messagebus/AbstractMessage.py
from adapt.messagebus.Message import Message
from adapt.messagebus.Envelope import Envelope

class AbstractMessage(Message):
    def __init__(self):
        self.envelope = Envelope()

    def getEnvelope(self):
        return self.envelope

    def setEnvelope(self, envelope):
        self.envelope = envelope

    def serialize(self):
        ret = self.__dict__.copy()
        if "envelope" in ret:
            del ret["envelope"]
        return ret
    def deserialize(self, dct):
        self.__dict__ = dct.copy()
